Failed to run the kubevirt  integration tests

**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
```
$ git rev-parse HEAD
a97e74308d061f0c38bad03c0c4dacbd3b4b0345
$ cd tests/
$ ls  default-config.json
default-config.json

$ go test -kubeconfig=/home/kni/dev-scripts/ocp/auth/kubeconfig
Running Suite: Tests Suite
==========================
Random Seed: 1563199102
Will run 408 of 409 specs

Panic [0.001 seconds]
[BeforeSuite] BeforeSuite
/home/kni/kubevirt/tests/tests_suite_test.go:44

  Test Panicked
  Couldn't load default test suite configuration: open tests/default-config.json: no such file or directory

  /usr/lib/golang/src/runtime/panic.go:513

  Full Stack Trace
  	/home/kni/go/pkg/mod/github.com/onsi/ginkgo@v1.8.0/internal/leafnodes/runner.go:107 +0x73
  panic(0x150ee80, 0xc000384220)
  	/usr/lib/golang/src/runtime/panic.go:513 +0x1b9
  kubevirt.io/kubevirt/tests.NewKubeVirtTestsConfiguration(0xc000384101)
  	/home/kni/kubevirt/tests/config.go:63 +0x10f
  kubevirt.io/kubevirt/tests.loadConfig(0xc000242300)
  	/home/kni/kubevirt/tests/config.go:70 +0x26
  kubevirt.io/kubevirt/tests.BeforeTestSuitSetup()
  	/home/kni/kubevirt/tests/utils.go:611 +0x67
  kubevirt.io/kubevirt/tests_test.glob..func18()
  	/home/kni/kubevirt/tests/tests_suite_test.go:45 +0x20
  github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc00091a7e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
  	/home/kni/go/pkg/mod/github.com/onsi/ginkgo@v1.8.0/internal/leafnodes/runner.go:113 +0x9c
  github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc00091a7e0, 0x1814c439, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
  	/home/kni/go/pkg/mod/github.com/onsi/ginkgo@v1.8.0/internal/leafnodes/runner.go:64 +0x12c
  github.com/onsi/ginkgo/internal/leafnodes.(*simpleSuiteNode).Run(0xc00030fae0, 0x1, 0x1, 0x0, 0x0, 0x0)
  	/home/kni/go/pkg/mod/github.com/onsi/ginkgo@v1.8.0/internal/leafnodes/suite_nodes.go:25 +0x8f
  github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runBeforeSuite(0xc000b172c0, 0x0)
  	/home/kni/go/pkg/mod/github.com/onsi/ginkgo@v1.8.0/internal/specrunner/spec_runner.go:123 +0x102
  github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).Run(0xc000b172c0, 0xb)
  	/home/kni/go/pkg/mod/github.com/onsi/ginkgo@v1.8.0/internal/specrunner/spec_runner.go:63 +0xb0
  github.com/onsi/ginkgo/internal/suite.(*Suite).Run(0xc00012c460, 0x7fc6ed6ae1b8, 0xc000b3a700, 0x1759523, 0xb, 0xc0003c41a0, 0x1, 0x1, 0x197b8c0, 0xc0000d8f40, ...)
  	/home/kni/go/pkg/mod/github.com/onsi/ginkgo@v1.8.0/internal/suite/suite.go:62 +0x28f
  github.com/onsi/ginkgo.RunSpecsWithCustomReporters(0x195a1a0, 0xc000b3a700, 0x1759523, 0xb, 0xc0003c4170, 0x1, 0x1, 0x1)
  	/home/kni/go/pkg/mod/github.com/onsi/ginkgo@v1.8.0/ginkgo_dsl.go:221 +0x247
  github.com/onsi/ginkgo.RunSpecsWithDefaultAndCustomReporters(0x195a1a0, 0xc000b3a700, 0x1759523, 0xb, 0xc000499f68, 0x0, 0x0, 0x7d1fdbf01b4b7d0)
  	/home/kni/go/pkg/mod/github.com/onsi/ginkgo@v1.8.0/ginkgo_dsl.go:209 +0xaa
  kubevirt.io/kubevirt/tests_test.TestTests(0xc000b3a700)
  	/home/kni/kubevirt/tests/tests_suite_test.go:41 +0xe4
  testing.tRunner(0xc000b3a700, 0x1812c78)
  	/usr/lib/golang/src/testing/testing.go:827 +0xbf
  created by testing.(*T).Run
  	/usr/lib/golang/src/testing/testing.go:878 +0x35c
```

**What you expected to happen**:
Test should run according to the documented steps at:
https://github.com/kubevirt/kubevirt/tree/master/tests#run-them-on-an-arbitrary-kubevirt-installation

**How to reproduce it (as minimally and precisely as possible)**:
$ go test -kubeconfig=/home/kni/dev-scripts/ocp/auth/kubeconfig


**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):
Server Version: version.Info{GitVersion:"v0.18.1", GitCommit:"4913335bc20764d0d0bec55da00146887726ae15", GitTreeState:"clean", BuildDate:"2019-06-13T10:22:49Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"4", Minor:"2+", GitVersion:"v4.2.0", GitCommit:"75945bd89", GitTreeState:"clean", BuildDate:"2019-05-31T13:22:16Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.0+5a5c44a", GitCommit:"5a5c44a", GitTreeState:"clean", BuildDate:"2019-07-03T13:03:58Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

> can you try to run `make functest`
```
make functest
hack/dockerized "hack/build-func-tests.sh"
STEP 1: FROM kubevirt/builder@sha256:08b4db91e5f6272004892c6174129172c1bef5c6906bde9f8414cb945a588e12
Getting image source signatures
Copying blob fbb0252de148 done
Copying blob 44821edabade done
Copying blob 730f49eeb0a7 done
Copying blob 012cb44e3253 done
Copying blob c015fae10118 done
Copying config 6f0b98680f done
Writing manifest to image destination
Storing signatures
STEP 2: ENV GIMME_GO_VERSION=1.11.5
8068b47608971dbf5ceda089d816b5a071fd0b34cca5592a3cfc5c4b8394cf0e
STEP 3: ENV GOPATH="/go" GOBIN="/usr/bin"
8648eaabf51154200449e185b323740157d0801aeeb167f43ace286ea85f5105
STEP 4: RUN     mkdir -p /go &&     source /etc/profile.d/gimme.sh &&     go get github.com/mattn/goveralls &&     go get -u github.com/golang/mock/gomock &&     go get -u github.com/rmohr/mock/mockgen &&     go get -u github.com/rmohr/go-swagger-utils/swagger-doc &&     go get -u github.com/onsi/ginkgo/ginkgo
go version go1.11.5 linux/amd64
9be363616309f5d1c026566ad14ea60eeda5730c0e3ed1a5f27fdce39a67cd15
STEP 5: RUN pip install j2cli && pip3 install operator-courier==1.3.0
WARNING: Running pip install with root privileges is generally not a good idea. Try `pip install --user` instead.
Collecting j2cli
  Downloading https://files.pythonhosted.org/packages/1b/81/930d5a727d43f8d2debcafeffbe725ff23d36668dbfb750e299d6813506b/j2cli-0.3.10-py2.py3-none-any.whl
Collecting jinja2>=2.7.2 (from j2cli)
  Downloading https://files.pythonhosted.org/packages/1d/e7/fd8b501e7a6dfe492a433deb7b9d833d39ca74916fa8bc63dd1a4947a671/Jinja2-2.10.1-py2.py3-none-any.whl (124kB)
    100% |################################| 133kB 1.0MB/s
Collecting MarkupSafe>=0.23 (from jinja2>=2.7.2->j2cli)
  Downloading https://files.pythonhosted.org/packages/fb/40/f3adb7cf24a8012813c5edb20329eb22d5d8e2a0ecf73d21d6b85865da11/MarkupSafe-1.1.1-cp27-cp27mu-manylinux1_x86_64.whl
Installing collected packages: MarkupSafe, jinja2, j2cli
Successfully installed MarkupSafe-1.1.1 j2cli-0.3.10 jinja2-2.10.1
WARNING: Running pip install with root privileges is generally not a good idea. Try `pip3 install --user` instead.
Collecting operator-courier==1.3.0
  Downloading https://files.pythonhosted.org/packages/55/dd/e9ea774c511d35984bc2763e30b6d236406634f946c8c34038b5fb457de1/operator-courier-1.3.0.tar.gz (346kB)
    100% |████████████████████████████████| 348kB 1.8MB/s
Collecting pyyaml (from operator-courier==1.3.0)
  Downloading https://files.pythonhosted.org/packages/a3/65/837fefac7475963d1eccf4aa684c23b95aa6c1d033a2c5965ccb11e22623/PyYAML-5.1.1.tar.gz (274kB)
    100% |████████████████████████████████| 276kB 1.6MB/s
Collecting requests (from operator-courier==1.3.0)
  Downloading https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl (57kB)
    100% |████████████████████████████████| 61kB 2.3MB/s
Collecting validators (from operator-courier==1.3.0)
  Downloading https://files.pythonhosted.org/packages/1d/96/8e386d1b8ef511ff10670ca98988c8f26e9d497216c569717fa2d3a9c06c/validators-0.13.0.tar.gz
Collecting semver (from operator-courier==1.3.0)
  Downloading https://files.pythonhosted.org/packages/21/18/a0de8cda637ba3efee1b3617ded00601507ce15bd70a39399740e0fd415f/semver-2.8.1-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests->operator-courier==1.3.0)
  Downloading https://files.pythonhosted.org/packages/69/1b/b853c7a9d4f6a6d00749e94eb6f3a041e342a885b87340b79c1ef73e3a78/certifi-2019.6.16-py2.py3-none-any.whl (157kB)
    100% |████████████████████████████████| 163kB 2.2MB/s
Collecting idna<2.9,>=2.5 (from requests->operator-courier==1.3.0)
  Downloading https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl (58kB)
    100% |████████████████████████████████| 61kB 2.8MB/s
Collecting chardet<3.1.0,>=3.0.2 (from requests->operator-courier==1.3.0)
  Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
    100% |████████████████████████████████| 143kB 2.9MB/s
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (from requests->operator-courier==1.3.0)
  Downloading https://files.pythonhosted.org/packages/e6/60/247f23a7121ae632d62811ba7f273d0e58972d75e58a94d329d51550a47d/urllib3-1.25.3-py2.py3-none-any.whl (150kB)
    100% |████████████████████████████████| 153kB 2.2MB/s
Requirement already satisfied: six>=1.4.0 in /usr/lib/python3.6/site-packages (from validators->operator-courier==1.3.0)
Collecting decorator>=3.4.0 (from validators->operator-courier==1.3.0)
  Downloading https://files.pythonhosted.org/packages/5f/88/0075e461560a1e750a0dcbf77f1d9de775028c37a19a346a6c565a257399/decorator-4.4.0-py2.py3-none-any.whl
Installing collected packages: pyyaml, certifi, idna, chardet, urllib3, requests, decorator, validators, semver, operator-courier
  Running setup.py install for pyyaml ... done
  Running setup.py install for validators ... done
  Running setup.py install for operator-courier ... done
Successfully installed certifi-2019.6.16 chardet-3.0.4 decorator-4.4.0 idna-2.8 operator-courier-1.3.0 pyyaml-5.1.1 requests-2.22.0 semver-2.8.1 urllib3-1.25.3 validators-0.13.0
8b70ee73ea51a16c7da059fa95c64661711fd79141212af4bd09bec57538d064
STEP 6: COPY rsyncd.conf /etc/rsyncd.conf
2fe37dec4d0837913e4a87984c3645786ee34b38a16a795ae62b53da9183f83f
STEP 7: ENTRYPOINT [ "/entrypoint.sh" ]
STEP 8: COMMIT kubevirt
3aaf3ba74116d35851dc7aa4f2725059670204db27c1d521e9d10a90d7dcc1b1
Error: unknown flag: --name
make: *** [functest] Error 125
```

> @gklein can you please move the kubevirt folder under.
> 
> `$GOPATH/src/kubevirt.io/` and try again

```
$ echo $GOPATH
/home/kni/go
$ mkdir -p $GOPATH/src/kubevirt.io/
$ cp -r kubevirt $GOPATH/src/kubevirt.io/
$ cd $GOPATH/src/kubevirt.io/kubevirt/tests
tests$ go test -kubeconfig=/home/kni/dev-scripts/ocp/auth/kubeconfig
Running Suite: Tests Suite
==========================
Random Seed: 1563275577
Will run 408 of 409 specs

Panic [0.001 seconds]
[BeforeSuite] BeforeSuite
/home/kni/go/src/kubevirt.io/kubevirt/tests/tests_suite_test.go:44

  Test Panicked
  Couldn't load default test suite configuration: open tests/default-config.json: no such file or directory

  /usr/lib/golang/src/runtime/panic.go:513

  Full Stack Trace
        /usr/lib/golang/src/runtime/panic.go:513 +0x1b9
  kubevirt.io/kubevirt/tests.NewKubeVirtTestsConfiguration(0x174e801)
        /home/kni/go/src/kubevirt.io/kubevirt/tests/config.go:63 +0x10f
  kubevirt.io/kubevirt/tests.loadConfig(0xc0005e2040)
        /home/kni/go/src/kubevirt.io/kubevirt/tests/config.go:70 +0x26
  kubevirt.io/kubevirt/tests.BeforeTestSuitSetup()
        /home/kni/go/src/kubevirt.io/kubevirt/tests/utils.go:611 +0x67
  kubevirt.io/kubevirt/tests_test.glob..func18()
        /home/kni/go/src/kubevirt.io/kubevirt/tests/tests_suite_test.go:45 +0x20
  kubevirt.io/kubevirt/vendor/github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc000652ea0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/kni/go/src/kubevirt.io/kubevirt/tests/tests_suite_test.go:41 +0xe4
  testing.tRunner(0xc000612300, 0x180d670)
        /usr/lib/golang/src/testing/testing.go:827 +0xbf
  created by testing.(*T).Run
        /usr/lib/golang/src/testing/testing.go:878 +0x35c

------------------------------

Waiting for namespace kubevirt-test-default to be removed, this can take a while ...
Waiting for namespace kubevirt-test-alternative to be removed, this can take a while ...

Ran 408 of 0 Specs in 8.708 seconds
FAIL! -- 0 Passed | 408 Failed | 0 Pending | 0 Skipped
--- FAIL: TestTests (8.71s)
FAIL
exit status 1
```

--- a/tests/README.md
+++ b/tests/README.md
@@ -8,7 +8,7 @@ point the tests to the cluster.
 
 ```
 cd tests # from the git repo root folder
-go test -kubeconfig=path/to/my/config
+go test -kubeconfig=path/to/my/config -config=default-config.json
 ```
 
 ## Run them on one of the core KubeVirt providers

--- a/tests/config.go
+++ b/tests/config.go
@@ -21,23 +21,9 @@ package tests
 
 import (
 	"encoding/json"
-	"flag"
-
-	"fmt"
 	"io/ioutil"
 	"os"
 )
-
-const (
-	DefaultConfigFile string = "tests/default-config.json"
-)
-
-var ConfigFile = ""
-var Config *KubeVirtTestsConfiguration
-
-func init() {
-	flag.StringVar(&ConfigFile, "config", "", "Path to a JSON formatted file from which the test suite will load its configuration. The path may be absolute or relative; relative paths start at the current working directory.")
-}
 
 // KubeVirtTestsConfiguration contains the configuration for KubeVirt tests
 type KubeVirtTestsConfiguration struct {
@@ -53,38 +39,11 @@ type KubeVirtTestsConfiguration struct {
 	StorageClassWindows string `json:"storageClassWindows"`
 }
 
-// Returns a new KubeVirtTestsConfiguration with default values
-func NewKubeVirtTestsConfiguration() *KubeVirtTestsConfiguration {
-	config := &KubeVirtTestsConfiguration{}
-
-	err := loadConfigFromFile(DefaultConfigFile, config)
-
-	if err != nil {
-		panic(fmt.Sprintf("Couldn't load default test suite configuration: %s\n", err))
-	}
-
-	return config
-}
-
-func loadConfig() *KubeVirtTestsConfiguration {
-	config := NewKubeVirtTestsConfiguration()
-
-	if ConfigFile != "" {
-		err := loadConfigFromFile(ConfigFile, config)
-
-		if err != nil {
-			panic(fmt.Sprintf("Couldn't load test suite configuration file: %s\n", err))
-		}
-	}
-
-	return config
-}
-
-func loadConfigFromFile(file string, config *KubeVirtTestsConfiguration) error {
+func loadConfig() (*KubeVirtTestsConfiguration, error) {
 	// open configuration file
-	jsonFile, err := os.Open(file)
+	jsonFile, err := os.Open(ConfigFile)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	defer jsonFile.Close()
@@ -92,8 +51,9 @@ func loadConfigFromFile(file string, config *KubeVirtTestsConfiguration) error {
 	// read the configuration file as a byte array
 	byteValue, _ := ioutil.ReadAll(jsonFile)
 
-	// convert the byte array as a KubeVirtTestsConfiguration struct
+	// convert the byte array to a KubeVirtTestsConfiguration struct
+	config := &KubeVirtTestsConfiguration{}
 	err = json.Unmarshal(byteValue, config)
 
-	return err
+	return config, err
 }

--- a/tests/utils.go
+++ b/tests/utils.go
@@ -98,6 +98,8 @@ var KubeVirtVirtctlPath = ""
 var KubeVirtInstallNamespace string
 var PreviousReleaseTag = ""
 var PreviousReleaseRegistry = ""
+var ConfigFile = ""
+var Config *KubeVirtTestsConfiguration
 
 var DeployTestingInfrastructureFlag = false
 var PathToTestingInfrastrucureManifests = ""
@@ -117,7 +119,7 @@ func init() {
 	flag.StringVar(&PathToTestingInfrastrucureManifests, "path-to-testing-infra-manifests", "manifests/testing", "Set path to testing infrastructure manifests")
 	flag.StringVar(&PreviousReleaseTag, "previous-release-tag", "", "Set tag of the release to test updating from")
 	flag.StringVar(&PreviousReleaseRegistry, "previous-release-registry", "", "Set registry of the release to test updating from")
-
+	flag.StringVar(&ConfigFile, "config", "tests/default-config.json", "Path to a JSON formatted file from which the test suite will load its configuration. The path may be absolute or relative; relative paths start at the current working directory.")
 }
 
 func FlagParse() {
@@ -608,7 +610,9 @@ func BeforeTestSuitSetup() {
 	log.InitializeLogging("tests")
 	log.Log.SetIOWriter(GinkgoWriter)
 
-	Config = loadConfig()
+	var err error
+	Config, err = loadConfig()
+	Expect(err).ToNot(HaveOccurred())
 
 	createNamespaces()
 	createServiceAccounts()

